frontend_v2/src/app/components/challenge/challengesubmit/challengesubmit.component.ts

```diff
@@ -230,6 +246,22 @@ isSubmissionUsingUrl: any;
     this.authToken = this.globalService.getAuthToken();
   }
 
+  progress(files: FileList) {
+    this.isFileSelected = true;
+    const fileToUpload = files.item(0);
+    const seconds = fileToUpload.size;
+    const timer$ = interval(10);
+
+    const sub = timer$.subscribe((sec) => {
+      this.progressValue = 0 + sec * 100 / seconds;
+      this.curSec = sec;
+
+      if (this.curSec === seconds) {
```

Comment:
Seems like it's also possible to unsubscribe even before the progressValue becoming 100. Will this condition become true for every upload?

Reply:
@lunayach as mentioned in the condition [here](https://github.com/Cloud-CV/EvalAI/pull/2832) The `curSec` value will be equal to the `seconds` value only when the `progressValue` becomes 100. And yes for every upload, this `seconds` will change based on the file size and will only be subscribed if the `progressValue` becomes 100.